### ISSUE TITLE
Buggy type conversion

### ISSUE THREAD

**Issue by ivan-kleshnin** [AmpersandJS/ampersand-input-view#31]
This is an input of type 'number' straight from the demo app.

```
new InputView({
  label: "Coolness Factor",
  name: "coolnessFactor",
  value: this.model && this.model.coolnessFactor,
  type: "number", 
})
```

If I focus input, enter some number and submit form it sends value as Number 
```
{"id":11, "firstName": "Ivan", "lastName": "Kleshnin", "coolnessFactor": 1}
```
Lets then rerender page. If I now submit form (omitting manual input action) it sends value as String!
```
{"id": 11, "firstName": "Ivan", "lastName": "Kleshnin", "coolnessFactor": "1"}
```
It looks like, Ampersand-Input somehow binds it type conversion with input action which is obviously wrong but may be it's just a side-effect of something else. Documentation mentions there is a possibility to make value derived:
```
 derived: {
  value: {
    deps: ['...'],
    fn: function () {
      ...
    }
},
```
but I want to get basic conventions on this. I'm used to think that types should be set in declarative way and I don't want to define those deps for *every* field.

**Comment by KasaiMagi** [comment]
I dont think this is actually where the issue is. If you're populated the value with a Number type it sets it correct in initialize(). The issue is in render() calling setValue().
```JavaScript
    setValue: function (value, skipValidation) {
        if (!value && value !== 0) {
            this.input.value = '';
        } else {
            this.input.value = value.toString();
        }
        this.inputValue = this.input.value;
        if (!skipValidation && !this.getErrorMessage()) {
            this.shouldValidate = true;
        }
    },
```
I'm not sure why the method is setting inputValue to the string value since when a user edits the field handleInputChange is called:
```JavaScript
    handleInputChanged: function () {
        if (document.activeElement === this.input) {
            this.directlyEdited = true;
        }
        this.inputValue = this.clean(this.input.value);
    },
```

**Comment by wraithgar** [comment]
This got fixed in 4.0.1
